proposals/csharp-9.0/native-integers.md

```diff
@@ -43,34 +43,39 @@ There is an identity conversion between compound types that differ by native int
 The tables below cover the conversions between special types.
 (The IL for each conversion includes the variants for `unchecked` and `checked` contexts if different.)
 
+General notes:
+- Most docs explain `conv.u` as conversion to __native unsigned integer__ and `conv.i` as conversion to __native integer__. But a more appropriate description is that `conv.u` is zero-extending and `conv.i` is sign-extending (if the most significant bit is `1`, then the conversion fills with `1`s).
+- Some non-obvious choices are guided by pre-existing semantics. For example, `sbyte->nint` uses `conv.i` and `byte->nint` uses `conv.u`, the same way that `sbyte->long` uses `conv.i8` and `byte->long` uses `conv.u8`. Also, `checked(void*->nint)` uses `conv.ovf.i.un` the same way that `checked(void*->long)` uses `conv.ovf.i8.un`.
+
 | Operand | Target | Conversion | IL |
 |:---:|:---:|:---:|:---:|
 | `object` | `nint` | Unboxing | `unbox` |
-| `void*` | `nint` | PointerToVoid | `conv.i` |
+| `void*` | `nint` | PointerToVoid | `conv.i`/`conv.ovf.i.un` |
```

Comment:
Minor: Considering adding spaces around `/` for consistency with the other cases.

Reply:
Why do we emit `conv.i` for `void*` to `nint`?

Assuming `void*` is treated as unsigned; then this should be similar to `uint to int` or `ulong to long` and therefore be a `nop`, correct?

You can see for https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs4fuwDsM6AWQAUAVz4CEASjoBeAHx1hEqQgDcHLrTq4IvAOZCxug3WlzFwk/tUbKXbnV4SjvcfzMyFS5/1ubOAMcXADkxF3NvMAALGDAAaxgAE2FlP2k7B21rOjDRHMjFGLjElKs9Gwyg7V8BMLcIryLYhOTU2tUpOwBfIA===:
* `unchecked uint to int` is a `nop`
* `unchecked ulong to long` is a `nop`
* `unchecked nuint to nint` is a `conv.i`
*
* `unchecked uint to int` is a `conv.ovf.i4.un`
* `unchecked ulong to long` is a `conv.ovf.i8.un`
* `unchecked nuint to nint` is a `conv.ovf.i.un`

So `nuint to nint` is the "inconsistent" thing.

Reply:
We indeed emit nop for `void*->nint`. I'll update the table to reflect.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/60944](https://github.com/dotnet/roslyn/issues/60944) for `nuint->nint` (should be `nop` instead of `conv.i`)